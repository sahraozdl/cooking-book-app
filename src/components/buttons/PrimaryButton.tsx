"use client";

import { Button } from "@headlessui/react";
import { type ButtonHTMLAttributes, type ReactNode } from "react";
import clsx from "clsx";

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
}

export default function PrimaryButton({ children, className, ...props }: Props) {
  return (
    <Button
      {...props}
      className={clsx(
        "px-4 py-2 rounded-full bg-orange-500 text-white hover:bg-orange-600 transition disabled:opacity-50 cursor-pointer focus:outline-orange-800",
        className
      )}
    >
      {children}
    </Button>
  );
}
