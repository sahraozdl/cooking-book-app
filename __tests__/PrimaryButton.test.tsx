import { render, screen, fireEvent } from "@testing-library/react";
import PrimaryButton from "@/components/buttons/PrimaryButton";

describe("PrimaryButton", () => {
  it("renders children correctly", () => {
    render(<PrimaryButton>Click Me</PrimaryButton>);
    expect(screen.getByText("Click Me")).toBeInTheDocument();
  });

  it("handles click events", () => {
    const handleClick = jest.fn();
    render(<PrimaryButton onClick={handleClick}>Click Me</PrimaryButton>);
    fireEvent.click(screen.getByText("Click Me"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<PrimaryButton>Snapshot</PrimaryButton>);
    expect(asFragment()).toMatchSnapshot();
  });
});
