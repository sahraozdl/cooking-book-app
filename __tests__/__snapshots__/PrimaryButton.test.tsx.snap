// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PrimaryButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-full bg-orange-500 text-white hover:bg-orange-600 transition disabled:opacity-50 cursor-pointer focus:outline-orange-800"
    data-headlessui-state=""
    type="button"
  >
    Snapshot
  </button>
</DocumentFragment>
`;
